cpp/include/cudf/groupby.hpp

```diff
@@ -68,6 +70,19 @@ struct aggregation_result {
   std::vector<std::unique_ptr<column>> results{};
 };
 
+/**
+ * @brief The result of a scan aggregation
+ *
+ * A `scan_result` contains the result of a groupby scan operation. `result`
+ * holds the result column that contains scanned result with each value in
+ * their sorted groups. Values of the same group locate in contiguous memory,
+ * whose order is stable. Order of groups is unguaranteed.
+ *
+ */
+struct scan_aggregation_result {
+  std::unique_ptr<column> result{};
```

Comment:
The purpose of `aggregation_result` was that it returns a `vector<column>` of all the results of all the aggregations for a single column. Wrapping a single `unique_ptr<column>` seems like unnecessary indirection.

Reply:
It's possible, for example a `scan_mean` can be achieved by a `scan_sum` and `scan_count`, so accepting multiple `scan_aggregation` requests can benefit from caching. If `scan_aggregation_result` holds a vector of columns, should it just be:

```c++
struct scan_aggregation_result:aggregation_result {};
```

Reply:
I don't think inheritance is necessary here and may cause more confusion than it's worth.